### PR TITLE
UKS-GGA: TDDFT and Stability Analysis

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
## Description
[My](https://github.com/psi4/psi4/pull/2824) [previous](https://github.com/psi4/psi4/pull/2832) [PRs](https://github.com/psi4/psi4/pull/2834) fixed the bugs making UKS TDDFT and stability analysis incorrect, so let's enable them.

## User API & Changelog headlines

- [x] Stability analysis is now supported for unrestricted GGAs.
- [x] TDDFT is now supported for unrestricted GGAs. 

## Dev notes & details
- [x] The prohibition against VV10 is not added to this PR, but its enforcement is. The old code would have given incorrect results.

## Checklist
- [x] Tests _mostly_ pass. `test_tdscf_excitations.py` has this PR covered. The one complication is that we're failing the TDA and RPA tests for CH2 with HCTH93. My investigations lead me to suspect the fault is bad reference data, probably involving pruned integration grids. @bgpeyton has confirmed pruning as the issue and volunteered to re-generate proper data.

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Comment by susilehtola** [comment]
I would consider not making VV10 throw an error, since it has negligible effects on the electron density. Najibi and Goerigk write in [doi:10.1021/acs.jctc.8b00842](https://doi.org/10.1021/acs.jctc.8b00842)

> While in a strictest sense, energy calculations with vdW-DFAs should be carried out fully self-consistently, we demonstrate conclusively for a total of 11 methods that such a strategy only increases the computational time eﬀort without having any signiﬁcant eﬀect on energetic properties, electron densities, or orbital-energy diﬀerences.

The recent analytical frequencies study by Feng, Liang and Head-Gordon in [arXiv:2208.14639](https://arxiv.org/abs/2208.14639) states

> The contribution of VV10 to simulating harmonic frequencies is shown to be small for small molecules but important for systems where weak interactions are important, such as water clusters.

That manuscript does have analytical second derivatives, so I guess one could extend it to TDDFT as well?

**Comment by JonathonMisiewicz** [comment]
Psi has a mechanism to make VV10 a post-iteration correction. How about allowing stability analysis / TDDFT if VV10 is post-iterative, and raising an error otherwise, directing users to make VV10 post-iterative?

**Comment by hokru** [comment]
I think it is a bit of an open questions how to exactly incorporate a VV10 dispersion treatment for excited states.

**Comment by JonathonMisiewicz** [comment]
Converted this PR to a draft. I do no support merging this PR while [a LibXC bug](https://gitlab.com/libxc/libxc/-/issues/445) is under investigation.

File: tests/pytests/test_tdscf_excitations.py [hunk 1]
```diff
@@ -86,6 +87,17 @@ def molecules():
...
+### On Signs
+# Different programs have different sign conventions for the various dipole moments.
+# These dipole moments are only defined up to a common phase, because they depend linearly on
+# the TDSCF eigenvectors, which are only defined up to phase.
+# The choice of sign is not important for _oscillator_strength, but _rotatory_strength
+# requires that the electric and magnetic dipole have the same phase.
+# In Psi's sign convention, the "L" gauge needs a + sign, and the "V" gauge a -
+# sign. Adjust for your case.
+# In PySCF's sign convention, the velocity gauge electric and the magnetic dipole
+# both have flipped sign.
```
Comment (loriab) [inline tests/pytests/test_tdscf_excitations.py:99]: very good to know

**Review by loriab — APPROVED** [review]
hooray!

File: psi4/driver/procrouting/response/scf_response.py [hunk 2]
```diff
@@ -402,9 +402,8 @@ def _validate_tdscf(*, wfn, states, triplets, guess) -> None:
     if restricted and wfn.functional().needs_xc() and do_triplets:
         raise ValidationError("TDSCF: Restricted Vx kernel only spin-adapted for singlets")
 
-    not_lda = wfn.functional().is_gga() or wfn.functional().is_meta()
-    if (not restricted) and not_lda:
-        raise ValidationError("TDSCF: Unrestricted Kohn-Sham Vx kernel currently limited to SVWN functional")
+    if wfn.functional().is_meta() or wfn.functional().needs_vv10():
+        raise ValidationError("TDSCF: Unrestricted Kohn-Sham Vx kernel does not support meta or VV10 functionals.")
```
Comment (JonathonMisiewicz) [inline psi4/driver/procrouting/response/scf_response.py:406]:
```suggestion
        raise ValidationError("TDSCF: Kohn-Sham Vx kernel does not support meta or VV10 functionals.")
```

**Review by jturney — APPROVED** [review]

**Pull request merged**